### PR TITLE
Add a test to verify that only one instance of stratisd can be running

### LINKED ISSUE #1644: stratisd has just enough action going on during startup that we need to put in some tests for that
These are not unit tests, and they are not dbus-tests, so we need to figure out how to construct them and where to place them.

### PR DESCRIPTION
**Pull request by mulkieran**
Closes #1644.

### CODE REVIEW

**Comment by mulkieran** [comment]
Had to allow an "unnecessary lambda"

File: tests/client-dbus/tests/misc/test_startup.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        self.first_process = subprocess.Popen(
+            self.command_line,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            close_fds=True,
+            env=os.environ,
+        )
+
+        # pylint: disable=unnecessary-lambda
+        self.addCleanup(lambda: self.first_process.terminate())
```
Comment (jbaublitz) [inline tests/client-dbus/tests/misc/test_startup.py]: Is the reason that this is:
```self.addCleanup(lambda: self.first_process.terminate())```
and not:
```self.addCleanup(self.first_process.terminate)```
just for readability purposes?

Comment (mulkieran) [reply tests/client-dbus/tests/misc/test_startup.py]: No, I think it's a hold over from a previous edit. I should be able to get rid of it. Let me give that a try.

**Comment by mulkieran** [comment]
rebased, and squashed removal of the pylint directive, see comment: https://github.com/stratis-storage/stratisd/pull/1643#pullrequestreview-294430733.

**Pull request merged**